src/solver/variable/include/antares/solver/variable/economy/minDispatchableGenByPlant.h

```diff
@@ -0,0 +1,281 @@
+    //! Indentation (GUI)
+    static constexpr uint8_t nodeDepthForGUI = +0;
+    //! Decimal precision
+    static constexpr uint8_t decimal = 0;
+    //! Number of columns used by the variable
+    static constexpr int columnCount = Category::dynamicColumns;
+    //! The Spatial aggregation
+    static constexpr uint8_t spatialAggregate = Category::spatialAggregateSum;
+    static constexpr uint8_t spatialAggregateMode = Category::spatialAggregateEachYear;
+    static constexpr uint8_t spatialAggregatePostProcessing = 0;
+    //! Intermediate values
+    static constexpr uint8_t hasIntermediateValues = 1;
+    //! Can this variable be non applicable (0 : no, 1 : yes)
+    static constexpr uint8_t isPossiblyNonApplicable = 0;
+
+    typedef IntermediateValues IntermediateValuesDeepType;
+    typedef std::vector<IntermediateValues> IntermediateValuesBaseType;
+    typedef std::vector<IntermediateValuesBaseType> IntermediateValuesType;
+
+}; // class VCard
+
+/*!
+** \brief Energy generated by all thermal dispatchable clusters
+*/
+template<class NextT = Container::EndOfList>
+class MinDispatchableGenByPlant: public Variable::IVariable<MinDispatchableGenByPlant<NextT>,
+                                                            NextT,
+                                                            VCardMinDispatchableGenByPlant>
+{
+public:
+    //! Type of the next static variable
+    typedef NextT NextType;
+    //! VCard
+    typedef VCardMinDispatchableGenByPlant VCardType;
+    //! Ancestor
+    typedef Variable::IVariable<MinDispatchableGenByPlant<NextT>, NextT, VCardType> AncestorType;
+
+    //! List of expected results
+    typedef typename VCardType::ResultsType ResultsType;
+
+    typedef VariableAccessor<ResultsType, VCardType::columnCount> VariableAccessorType;
+
+    enum
+    {
+        //! How many items have we got
+        count = 1 + NextT::count,
+    };
+
+    template<int CDataLevel, int CFile>
+    struct Statistics
+    {
+        enum
+        {
+            count = ((VCardType::categoryDataLevel & CDataLevel
+                      && VCardType::categoryFileLevel & CFile)
+                       ? (NextType::template Statistics<CDataLevel, CFile>::count
+                          + VCardType::columnCount * ResultsType::count)
+                       : NextType::template Statistics<CDataLevel, CFile>::count),
+        };
+    };
+
+public:
+    MinDispatchableGenByPlant() = default;
+    ~MinDispatchableGenByPlant() = default;
+
+    void initializeFromArea(Data::Study* study, Data::Area* area)
+    {
+        // Get the number of years in parallel
+        pNbYearsParallel = study->maxNbYearsInParallel;
+        pValuesForTheCurrentYear.resize(pNbYearsParallel);
+
+        // Get the area
+        nbClusters_ = std::ranges::count_if(area->thermal.list.all(),
+                                            [](auto c)
+                                            { return c->isEnabled() && !c->isMustRun(); });
```

Comment:
Please be consistent with other "by plant" variables (e.g Profit by plant, DTG by plant, etc.)

```cpp
        pSize = area->thermal.list.enabledCount();
```